pkg/services/udp_proxy_test.go

```diff
@@ -150,3 +153,111 @@ func TestUDPProxyServiceOutbound(t *testing.T) {
+	tests := []struct {
+		name           string
+		expectedAddr   netceptor.Addr
+		calls          func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface)
+		expectContinue bool
+	}{
+		{
+			name:         "ReadFrom error",
+			expectedAddr: expectedAddr,
+			calls: func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface) {
+				mockPacketConner.EXPECT().ReadFrom(gomock.Any()).Return(0, expectedUDPAddr, fmt.Errorf("ReadFrom error")).Times(1)
+				mockPacketConner.EXPECT().GetLogger().Return(logger.NewReceptorLogger("")).Times(1)
+			},
+		},
+		{
+			name:         "Expected Address mismatch",
+			expectedAddr: expectedAddr,
+			calls: func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface) {
+				mockPacketConner.EXPECT().ReadFrom(gomock.Any()).Return(0, unexpectedUDPAddr, nil).Times(1)
+				mockPacketConner.EXPECT().GetLogger().Return(logger.NewReceptorLogger("")).Times(1)
+			},
+		},
+		{
+			name:         "WriteTo error",
+			expectedAddr: expectedAddr,
+			calls: func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface) {
+				mockPacketConner.EXPECT().ReadFrom(gomock.Any()).Return(1, expectedUDPAddr, nil).Times(1)
+				mockUDPConnInterface.EXPECT().WriteTo(gomock.Any(), gomock.Any()).Return(1, fmt.Errorf("WriteTo error")).Times(1)
+				mockPacketConner.EXPECT().GetLogger().Return(logger.NewReceptorLogger("")).Times(1)
+			},
+		},
+		{
+			name:         "Written bytes mismatch",
+			expectedAddr: expectedAddr,
+			calls: func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface) {
+				mockPacketConner.EXPECT().ReadFrom(gomock.Any()).Return(7, expectedUDPAddr, nil).Times(1)
+				mockUDPConnInterface.EXPECT().WriteTo(gomock.Any(), gomock.Any()).Return(5, nil).Times(1)
+				mockPacketConner.EXPECT().GetLogger().Return(logger.NewReceptorLogger("")).Times(1)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+			mockPacketConner := mock_netceptor.NewMockPacketConner(ctrl)
+			mockUDPConnInterface := mock_net_interface.NewMockUDPConnInterface(ctrl)
+			tt.calls(mockPacketConner, mockUDPConnInterface)
+			buf := make([]byte, utils.NormalBufferSize)
+			processInboundPacket(mockPacketConner, mockUDPConnInterface, expectedUDPAddr, tt.expectedAddr, buf)
+		})
+	}
+}
+
+func TestProcessOutboundPacket(t *testing.T) {
+	destinationAddr := &net.UDPAddr{IP: net.ParseIP("192.168.0.1"), Port: 2222}
+	tests := []struct {
+		name           string
+		Addr           netceptor.Addr
+		calls          func(mockPacketConner *mock_netceptor.MockPacketConner, mockUDPConnInterface *mock_net_interface.MockUDPConnInterface)
```

Comment:
```suggestion
		calls          func()
```
Same as above. Let's simplify to `func()` here.

Reply:
Done, thank you!